js/frontend/src/components/NewUrlForm/NewUrlFormContainer.tsx

```diff
@@ -20,32 +21,21 @@ export function NewUrlFormContainer() {
 
   const onSubmit: SubmitHandler<NewUrlFormSchema> = (values) => {
     const { url, miniUrl } = values;
-    const response = new Promise<{ data: { short_url: string } }>(
-      (resolve, reject) => {
-        axios
-          .post("/api/url/new", {
-            url: url,
-            short: miniUrl,
-          })
-          .then((res) => {
-            resolve(res);
-          })
-          .catch((err) => {
-            reject(err);
-          });
-      },
-    );
+    const response: NewUrlRequest = {
+      url: url,
+    };
+    if (miniUrl && miniUrl !== undefined) {
+      response.short = miniUrl;
+    }
```

Comment:
I think we'll want to update this constant- now that you expanded the type for `NewUrlRequest`, I don't think you have to handle `miniUrl && miniUrl !== undefined`

```suggestion
    const requestParams: NewUrlRequest = {
      url,
      short: miniUrl
    };
```

Reply:
Separately, I suggest we just update schema `short` -> `miniUrl`

Reply:
Updated!